diff --git a/etc/rancid.conf.sample.in b/etc/rancid.conf.sample.in
index 398d0dd..9080dcd 100644
--- a/etc/rancid.conf.sample.in
+++ b/etc/rancid.conf.sample.in
@@ -22,7 +22,7 @@ umask 027
 TMPDIR=/tmp; export TMPDIR
 # Be careful changing this, it affects CVSROOT below.  It should be a FQPN, not
 # relative.
-BASEDIR=@localstatedir@; export BASEDIR
+BASEDIR=@localstatedir@/lib/rancid; export BASEDIR
 PATH=@bindir@:@ENV_PATH@; export PATH
 # Location of the CVS/SVN repository.  Be careful changing this.
 # If RCSSYS is svn, this can be:
@@ -39,7 +39,7 @@ LOGDIR=$BASEDIR/logs; export LOGDIR
 # Select which RCS system to use, "cvs" (default) or "svn".  Do not change
 # this after CVSROOT has been created with rancid-cvs.  Changing between these
 # requires manual conversions.
-RCSSYS=@RCSSYS@; export RCSSYS
+RCSSYS=git; export RCSSYS
 #
 # if ACLSORT is NO, access-lists will NOT be sorted.
 #ACLSORT=YES; export ACLSORT
